packages/governance/src/types.js

```diff
@@ -87,79 +151,124 @@
+ * @property {VoteCounterCloseFacet} closeFacet
  */
 
 /**
- * @typedef {Object} CompleteWeightedBallot
- * @property {string} question
- * @property {Record<string,bigint>[]} weighted - list of positions with weights.
- *   BallotCounter may limit weights to a range or require uniqueness.
+ * @callback BuildVoteCounter
+ * @param {QuestionSpec} questionSpec
+ * @param {bigint} threshold - questionSpec includes quorumRule; the registrar
+ *    converts that to a number that the counter can enforce.
+ * @param {Instance} instance
+ * @returns {VoteCounterFacets}
  */
 
 /**
- * @typedef {Object} CompleteOrderedBallot
- * @property {string} question
- * @property {string[]} ordered - ordered list of position from most preferred to
- *   least preferred
+ * @callback LooksLikeQuestionSpec
+ * @param {unknown} allegedQuestionSpec
+ * @returns {QuestionSpec}
+ */
+
+/**
+ * @callback LooksLikeParamChangeIssue
+ * @param {unknown} issue
+ * @returns { asserts issue is ParamChangeIssue }
+ */
+
+/**
+ * @callback LooksLikeIssueForType
+ * @param {ElectionType} electionType
+ * @param {unknown} issue
+ * @returns { asserts issue is Issue }
+ */
+
+/**
+ * @callback LooksLikeSimpleIssue
+ * @param {unknown} issue
+ * @returns { asserts issue is SimpleIssue }
  */
 
 /**
- * @typedef { CompleteEqualWeightBallot | CompleteOrderedBallot | CompleteWeightedBallot } CompletedBallot
+ * @callback LooksLikeClosingRule
+ * @param {unknown} closingRule
+ * @returns { asserts closingRule is ClosingRule }
  */
 
 /**
  * @callback SubmitVote
- * @param {Handle<'Voter'>} seat
- * @param {ERef<CompletedBallot>} filledBallot
+ * @param {Handle<'Voter'>} voterHandle
+ * @param {Position[]} chosenPositions
  * @param {bigint=} weight
  */
 
 /**
- * @typedef {Object} VoterFacet
- * @property {SubmitVote} submitVote
+ * @typedef {Object} RegistrarPublic
+ * @property {() => Subscription<QuestionDetails>} getQuestionSubscription
+ * @property {() => Promise<Handle<'Question'>[]>} getOpenQuestions,
+ * @property {() => string} getName
+ * @property {() => Instance} getInstance
+ * @property {(h: Handle<'Question'>) => Promise<Question>} getQuestion
+ */
+
+/**
+ * @typedef {Object} PoserFacet
+ * @property {AddQuestion} addQuestion
+ */
+
+/**
+ * @typedef {Object} RegistrarCreatorFacet
+ *  addQuestion() can be used directly when the creator doesn't need any
+ *  reassurance. When someone needs to connect addQuestion to the Registrar
+ *  instance, getPoserInvitation() lets them get addQuestion with assurance.
+ * @property {() => Promise<Invitation>} getPoserInvitation
+ * @property {AddQuestion} addQuestion
+ * @property {() => Promise<Invitation>[]} getVoterInvitations
+ * @property {() => Subscription<QuestionDetails>} getQuestionSubscription
+ * @property {() => RegistrarPublic} getPublicFacet
  */
 
 /**
  * @typedef {Object} ClosingRule
- * @property {Timer} timer
+ * @property {ERef<Timer>} timer
  * @property {Timestamp} deadline
```

Comment:
as we just discovered, `deadline`s are `RelativeTime`s.

```suggestion
 * @property {ERef<Timer>} timer
 * @property {RelativeTime} deadline
```